BinDays.Api.Collectors/Collectors/Councils/NewForestDistrictCouncil.cs

```
@@ -0,0 +1,448 @@
+					{"PAGE:X", "0"},
+					{"PAGE:Y", "0"},
+					{"PAGE:F", "CTID-EBTnjgwK-_"},
+					{"ufsEndUser*", "1"},
+					{"pageSeq", "1"},
+					{"pageId", "Page_1"},
+					{"formStateId", "1"},
+					{"$USERVAR1", ""},
+					{"$USERVAR2", ""},
+					{"$USERVAR3", ""},
+					{"CTRL:JmLqCKl2:_:A", postcode},
+					{"CTRL:JmLqCKl2:_:B.h", ""},
+					{"HID:inputs", "ICTRL:JmLqCKl2:_:A,ACTRL:JmLqCKl2:_:B.h,ACTRL:EBTnjgwK:_,APAGE:E.h,APAGE:B.h,APAGE:N.h,APAGE:S.h,APAGE:R.h"},
+					{"CTRL:EBTnjgwK:_", "Submit"}
+				});
+
+				var requestHeaders = new Dictionary<string, string>() {
+					{"user-agent", Constants.UserAgent},
+					{"content-type", "application/x-www-form-urlencoded"},
+					{"cookie", cookie},
+				};
+
+				var clientSideRequest = new ClientSideRequest()
+				{
+					RequestId = 3,
+					Url = "https://forms.newforest.gov.uk/ufs/ufsajax?ebz=" + token,
+					Method = "POST",
+					Headers = requestHeaders,
+					Body = requestBody,
+				};
+
+				var getAddressesResponse = new GetAddressesResponse()
+				{
+					Addresses = null,
+					NextClientSideRequest = clientSideRequest
+				};
+
+				return getAddressesResponse;
+			}
+			// Process addresses from response
+			else if (clientSideResponse.RequestId == 3)
+			{
+				// Parse the JSON response
+				using var jsonDoc = JsonDocument.Parse(clientSideResponse.Content);
+
+				// Find the first html key that contains <option
+				string? htmlWithOptions = null;
+				if (jsonDoc.RootElement.TryGetProperty("updatedControls", out var updatedControls))
+				{
+					foreach (var control in updatedControls.EnumerateArray())
+					{
+						if (control.TryGetProperty("html", out var htmlProperty) &&
+							htmlProperty.ValueKind == JsonValueKind.String &&
+							htmlProperty.GetString()?.Contains("<option") == true)
+						{
+							htmlWithOptions = htmlProperty.GetString().Replace("\\\"", "\"");
+							break;
+						}
+					}
+				}
+				// TODO Error handling for when htmlWithOptions is null

```

When lookup fails, how do you want errors handled? Further down I threw an exception...

---

The current design is to just let exceptions be raised whether it be for nulls or something else.

Errors get captured and logged at a higher level to then be fixed. They are brittle by design and as each request is handled in descrete stages it is generally clear where something went wrong.

See other councils for how they don't handle exceptions.